### PR TITLE
[Angular] Fix strict in account and audits module

### PR DESCRIPTION
**Pull request by kaidohallik**
* after merging 3 PR (account, audits, core+tracker), strict option reported some problems, which this PR fixes
* also using new `getCurrentLanguage()` from the new `ng-jhipster`
* and also cleans code

Related to #10631 #10639 #10683 #10786

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/test/javascript/spec/app/account/settings/settings.component.spec.ts.ejs [hunk 1]
```diff
@@ -58,7 +47,7 @@ describe('Component Tests', () => {
         beforeEach(() => {
             fixture = TestBed.createComponent(SettingsComponent);
             comp = fixture.componentInstance;
-            mockAuth = fixture.debugElement.injector.get(AccountService);
+            mockAuth = (fixture.debugElement.injector.get(AccountService) as unknown) as MockAccountService;
```
Comment (wmarques) [inline generators/client/templates/angular/src/test/javascript/spec/app/account/settings/settings.component.spec.ts.ejs]: Can't we use TestBed.get(AccountService) ?

Comment (kaidohallik) [reply generators/client/templates/angular/src/test/javascript/spec/app/account/settings/settings.component.spec.ts.ejs]: Great hint, thanks! I made change and force pushed.

**Comment by pascalgrimaud** [comment]
@kaidohallik : is there a specific ticket for these kind of PR ? I saw a lot of PR like this and as it's a lot of work, I'd like to put a bounty for this

**Comment by kaidohallik** [comment]
As coming layouts PR is related to these components then I'm merging this.

**Pull request merged**